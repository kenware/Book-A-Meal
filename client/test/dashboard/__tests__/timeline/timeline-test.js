import React from 'react';
import { shallow } from 'enzyme';
import Timeline from '../../../../components/dashboard/timeline';
import { props } from '../../dashboardMockData';

describe('Test Timeline component of dashboard', () => {
  it('renders correctly', () => {
    const tree = shallow(<Timeline {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should have three modal-order-content class', () => {
    global.localStorage = {
      getItem() {
        return 'user';
      }
    };
    const tree = shallow(<Timeline {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
