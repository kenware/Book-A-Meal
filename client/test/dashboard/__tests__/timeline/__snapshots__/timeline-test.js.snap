// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Timeline component of dashboard renders correctly 1`] = `
<ul
  className="timeline"
>
  <li>
    <div
      className="timeline-badge"
    >
      <em
        className="fa fa-camera"
      />
    </div>
    <div
      className="timeline-panel bg-light"
    >
      <div
        className="timeline-heading"
      >
        <h5
          className="timeline-title l-margin m-text"
        >
          Profile
        </h5>
      </div>
      <div
        className="timeline-body"
      >
        <ul>
          <li
            style={
              Object {
                "listStyle": "none",
              }
            }
          >
            <img
              alt="profil"
              className="user-img rounded-circle"
            />
          </li>
          <li>
            <span
              className="h2-color"
            >
              Keny
            </span>
          </li>
          <li>
            <span
              className="h2-color"
            >
              Update profile
            </span>
          </li>
          <li>
            <span
              className="h2-color"
              role="button"
            >
               Change Password
            </span>
          </li>
          <span />
        </ul>
      </div>
    </div>
  </li>
  <li>
    <div
      className="timeline-badge primary"
    >
      <em
        className="fa fa-link"
      />
    </div>
    <div
      className="timeline-panel bg-light"
    >
      <div
        className="timeline-heading"
      >
        <h5
          className="timeline-title l-margin m-text"
        >
          Notifications
        </h5>
      </div>
      <div
        className="timeline-body"
      >
        <ul
          style={
            Object {
              "height": "300px",
              "overflow": "scroll",
            }
          }
        >
          <li
            key="1"
            style={
              Object {
                "marginTop": "1rem",
              }
            }
          >
            <div
              className="p-color"
            >
              here i come
            </div>
            <div>
              Jun
               
              10
                
              2018
            </div>
          </li>
        </ul>
      </div>
    </div>
  </li>
  <li>
    <div
      className="timeline-badge"
    >
      <em
        className="fa fa-paperclip"
      />
    </div>
    <div
      className="timeline-panel bg-light"
    >
      <div
        className="timeline-heading"
      >
        <h5
          className="timeline-title m-text l-margin"
        >
          Recently Ordered Meal
        </h5>
      </div>
      <div
        className="timeline-body"
      >
        <ul>
          <li />
          <li>
            Italian Recipes
          </li>
          <li>
            Ewedu Meal
          </li>
          <li>
            Abacha
          </li>
        </ul>
      </div>
    </div>
  </li>
</ul>
`;

exports[`Test Timeline component of dashboard should have three modal-order-content class 1`] = `
<ul
  className="timeline"
>
  <li>
    <div
      className="timeline-badge"
    >
      <em
        className="fa fa-camera"
      />
    </div>
    <div
      className="timeline-panel bg-light"
    >
      <div
        className="timeline-heading"
      >
        <h5
          className="timeline-title l-margin m-text"
        >
          Profile
        </h5>
      </div>
      <div
        className="timeline-body"
      >
        <ul>
          <li
            style={
              Object {
                "listStyle": "none",
              }
            }
          >
            <img
              alt="profil"
              className="user-img rounded-circle"
              src="user"
            />
          </li>
          <li>
            <span
              className="h2-color"
            >
              Keny
            </span>
          </li>
          <li>
            <span
              className="h2-color"
            >
              Update profile
            </span>
          </li>
          <li>
            <span
              className="h2-color"
              role="button"
            >
               Change Password
            </span>
          </li>
          <li>
            <button
              className="p-color"
            >
               Upgrade To A Caterer
            </button>
          </li>
        </ul>
      </div>
    </div>
  </li>
  <li>
    <div
      className="timeline-badge primary"
    >
      <em
        className="fa fa-link"
      />
    </div>
    <div
      className="timeline-panel bg-light"
    >
      <div
        className="timeline-heading"
      >
        <h5
          className="timeline-title l-margin m-text"
        >
          Notifications
        </h5>
      </div>
      <div
        className="timeline-body"
      >
        <ul
          style={
            Object {
              "height": "300px",
              "overflow": "scroll",
            }
          }
        >
          <li
            key="1"
            style={
              Object {
                "marginTop": "1rem",
              }
            }
          >
            <div
              className="p-color"
            >
              here i come
            </div>
            <div>
              Jun
               
              10
                
              2018
            </div>
          </li>
        </ul>
      </div>
    </div>
  </li>
  <li>
    <div
      className="timeline-badge"
    >
      <em
        className="fa fa-paperclip"
      />
    </div>
    <div
      className="timeline-panel bg-light"
    >
      <div
        className="timeline-heading"
      >
        <h5
          className="timeline-title m-text l-margin"
        >
          Recently Ordered Meal
        </h5>
      </div>
      <div
        className="timeline-body"
      >
        <ul>
          <li />
          <li>
            Italian Recipes
          </li>
          <li>
            Ewedu Meal
          </li>
          <li>
            Abacha
          </li>
        </ul>
      </div>
    </div>
  </li>
</ul>
`;
